src/schemathesis/specs/openapi/negative/mutations.py

```diff
@@ -0,0 +1,153 @@
+"""Schema mutations."""
+import enum
+from typing import Callable, Tuple
+
+from hypothesis import strategies as st
+from hypothesis.strategies._internal.featureflags import FeatureStrategy
+from hypothesis_jsonschema._canonicalise import canonicalish
+
+from .types import Draw, Schema
+from .utils import get_type
+
+
+class MutationResult(enum.Enum):
+    """The result of applying some mutation to some schema.
+
+    Failing to mutate something means that by applying some mutation, it is not possible to change
+    the schema in the way, so it covers inputs not covered by the "positive" strategy.
+
+    Knowing this, we know when the schema is mutated and whether we need to apply more mutations.
+    """
+
+    SUCCESS = 1
+    FAILURE = 2
+
+
+Mutation = Callable[[Draw, Schema], MutationResult]
+
+
+def remove_required_property(draw: Draw, schema: Schema) -> MutationResult:
+    """Remove a required property.
+
+    Effect: Some property won't be generated.
+    Applicable types: object
+    """
+    required = schema.get("required")
+    if not required:
+        # No required properties - can't mutate
+        return MutationResult.FAILURE
+    # TODO: apply swarm testing here
+    property_name = draw(st.sampled_from(sorted(required)))
+    required.remove(property_name)
+    if not required:
+        # In JSON Schema Draft 4, `required` must contain at least one string
+        # To keep the schema conformant, remove the `required` key completely
+        del schema["required"]
+    # An optional property still can be generated, and to avoid it, we need to remove it from other keywords.
+    properties = schema.get("properties", {})
+    properties.pop(property_name, None)
+    # This property still can be generated via `patternProperties`, but this implementation doesn't cover this case
+    # Its probability is relatively low, and the complete solution compatible with Draft 4 will require extra complexity
+    # The output filter covers cases like this
+    return MutationResult.SUCCESS
+
+
+def change_schema_type(draw: Draw, schema: Schema) -> MutationResult:
+    """Change type of values accepted by a schema.
+
+    Applicable types: any
+    """
+    if "type" not in schema:
```

Comment:
When would this happen?

Reply:
It turned out that underspecified schemas are pretty common, not saying that it happens too often, but it is a visible fraction. Usually, I see cases like `{"format": "date"}` or `{"properties": {...}}` which are perfectly valid JSON Schemas.